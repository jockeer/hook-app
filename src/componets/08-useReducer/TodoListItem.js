import React from 'react';

const TodoListItem = ({todo, i, handleToggle, handleDelete}) => {
    return (  
        <li 
            className="list-group-item">
            <p className={`${ todo.done && 'complete' }`} onClick={()=>handleToggle(todo.id)}>{i+1}.- {todo.desc}</p> 
            <button type='button' className="btn btn-danger" onClick={ () => handleDelete(todo.id)}>Borrar</button>
        </li>
    );
}
 
export default TodoListItem;