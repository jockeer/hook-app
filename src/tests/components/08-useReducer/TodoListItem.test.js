import React from 'react';
 
import {shallow} from 'enzyme'
import TodoListItem from '../../../componets/08-useReducer/TodoListItem';
import { demoTodos } from '../../fixtures/demoTodos';

describe('Prueba TodoListItem', () => { 
    
    const handleToggle = jest.fn()
    const handleDelete = jest.fn()
    const wrapper = shallow(
        <TodoListItem 
            todo={demoTodos[0]} 
            i={1} handleToggle={handleToggle} 
            handleDelete={handleDelete} 
        />
    )
    
    test('Debe mostrarse correctamente', () => { 

        expect(wrapper).toMatchSnapshot()
    })

    test('Debe llamar la funcion borrar', () => { 

        wrapper.find('button').simulate('click')

        expect(handleDelete).toHaveBeenCalled()
        expect(handleDelete).toHaveBeenCalledWith(expect.any(Number))
        
    })

    test('Debe llamar la funcion toogle', () => { 

        wrapper.find('p').simulate('click')

        expect(handleToggle).toHaveBeenCalledWith(expect.any(Number))
        
    })

    test('Debe mostrar el texto correctamente', () => { 

        const texto = wrapper.find('p').text().trim()

        expect(texto).toBe(`2.- ${demoTodos[0].desc}`)
        
    })

    test('Debe tener la clase complete si el TODO.done = true', () => { 

        const todo = demoTodos[0]
        todo.done = true

        const wrapper = shallow(
            <TodoListItem 
                todo={todo}
            />
        )
        expect(wrapper.find('p').hasClass('complete')).toBe(true)
        
    })

})