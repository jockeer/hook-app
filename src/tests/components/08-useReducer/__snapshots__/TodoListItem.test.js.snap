// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba TodoListItem Debe mostrarse correctamente 1`] = `
<li
  className="list-group-item"
>
  <p
    className="false"
    onClick={[Function]}
  >
    2
    .- 
    Aprender React
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
    type="button"
  >
    Borrar
  </button>
</li>
`;
